src/evaluate.js

```diff
@@ -56,28 +56,30 @@ const table = {
     // Evaluate special form such as `if` or `lambda`
     if (first.type === 'Symbol' && special.hasOwnProperty(first.name)) {
       return special[first.name](stack, scope, node);
+    }
     // Regular function call
-    } else {
-      const f = EVAL(stack, scope, first);
-      const args = node
-        .items
-        .slice(1)
-        .map(x => EVAL(stack, scope, x));
-      if (f.type === 'JSFunction') {
-        const args2 = [stack, scope].concat(args);
-        return f.f.apply(null, args2);
-      } else if (f.type === 'Function') {
-        const newStack = stack.concat(['#<lambda>']);
-        const obj = {};
-        f.parameters.items.forEach((p, i) => {
-          obj[p.name] = args[i];
-        });
-        const newScope = Scope.create(f.scope, obj);
-        const values = f.body.map(x => EVAL(newScope, newScope, x));
-        return values[values.length - 1];
-      }
+    const f = EVAL(stack, scope, first);
+    if (f.type !== 'JSFunction' && f.type !== 'Function') {
       throw new Error('cannot call non-function');
     }
+    const args = node
+      .items
+      .slice(1)
+      .map(x => EVAL(stack, scope, x));
+    if (f.type === 'JSFunction') {
+      const args2 = [stack, scope].concat(args);
+      return f.f.apply(null, args2);
+    }
+    if (f.type === 'Function') {
```

Comment:
I would either keep the `else if` here or switch to a dispatch table. I've had to deal with too many long functions with `if` and no `else` and it's not clear if there's even a `return` down each branch.